src/python/pants/option/global_options.py

```diff
@@ -81,14 +81,25 @@ class ExecutionOptions:
     remote_execution_process_cache_namespace: Any
     remote_instance_name: Any
     remote_ca_certs_path: Any
-    remote_oauth_bearer_token_path: Any
     remote_execution_extra_platform_properties: Any
-    remote_execution_headers: Any
+    remote_execution_headers: Union[None, List[Tuple[str, str]], Callable[[Optional[str]], dict]]
     remote_execution_overall_deadline_secs: int
     process_execution_local_enable_nailgun: bool
 
     @classmethod
     def from_bootstrap_options(cls, bootstrap_options):
+        headers = bootstrap_options.remote_execution_headers
+        if bootstrap_options.remote_oauth_bearer_token_path is not None:
+            with open(bootstrap_options.remote_oauth_bearer_token_path) as f:
+                token = f.read().strip()
+            headers["authorization"] = f"Bearer {token}"
+
+        headers_as_list = [(k, v) for (k, v) in headers.items()]
+
+        def get_headers(build_id: Optional[str]) -> List[Tuple[str, str]]:
+            print(f"get_headers: build_id={build_id}")
```

Comment:
leftovers?

Reply:
Eventually it'll be a leftover but for now I put it that to show that it actually did invoke the function. Will remove in next iteration.